datasets/lambada/lambada.py

```diff
@@ -0,0 +1,134 @@
+  month     = {August},
+  year      = {2016},
+  address   = {Berlin, Germany},
+  publisher = {Association for Computational Linguistics},
+  pages     = {1525--1534},
+  url       = {http://www.aclweb.org/anthology/P16-1144}
+}
+"""
+
+_DESCRIPTION = """
+The LAMBADA evaluates the capabilities of computational models
+for text understanding by means of a word prediction task.
+LAMBADA is a collection of narrative passages sharing the characteristic
+that human subjects are able to guess their last word if
+they are exposed to the whole passage, but not if they
+only see the last sentence preceding the target word.
+To succeed on LAMBADA, computational models cannot
+simply rely on local context, but must be able to
+keep track of information in the broader discourse.
+
+The LAMBADA dataset is extracted from BookCorpus and
+consists of 10'022 passages, divided into 4'869 development
+and 5'153 test passages. The training data for language
+models to be tested on LAMBADA include the full text
+of 2'662 novels (disjoint from those in dev+test),
+comprising 203 million words.
+"""
+
+_URL = "https://zenodo.org/record/2630551/files/lambada-dataset.tar.gz"
+
+
+class Lambada(datasets.GeneratorBasedBuilder):
+    """LAMBADA dataset."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="plain_text",
+            description="Plain text",
+            version=VERSION,
+        )
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "domain": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://zenodo.org/record/2630551#.X8UP76pKiIa",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "train-novels.tar"),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": os.path.join(data_dir, "lambada_test_plain_text.txt"), "split": "test"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "lambada_development_plain_text.txt"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        if split == "train":
```

Comment:
I think we usually want to extract in the `_split_generators` function. I think it's better to just hand over the `filepath` to `_generate_examples()` and then start with 

```python
if split == "train":
            recursion_pattern = f"{os.path.dirname(filepath)}/train-novels/*/*.txt"
            for idx, novel_path in enumerate(glob.iglob(recursion_pattern, recursive=True)):
```

Reply:
ok, let me move the un-tar in the `_split_generators`